src/git-utils.ts

```diff
@@ -17,18 +17,29 @@ export async function copyAssets(
   publishDir: string,
   workDir: string
 ): Promise<void> {
-  const copyOpts = {recursive: true, force: true};
-  const files = fs.readdirSync(publishDir);
-  core.debug(`${files}`);
-  for await (const file of files) {
-    if (file.endsWith('.git') || file.endsWith('.github')) {
-      continue;
+  if (process.platform === 'win32') {
+    const copyOpts = {recursive: true, force: true};
+    const files = fs.readdirSync(publishDir);
+    core.debug(`${files}`);
+    for await (const file of files) {
+      if (file.endsWith('.git') || file.endsWith('.github')) {
+        continue;
+      }
+      const filePath = path.join(publishDir, file);
+      await io.cp(filePath, `${workDir}/`, copyOpts);
+      core.info(`[INFO] copy ${file}`);
     }
-    const filePath = path.join(publishDir, file);
-    await io.cp(filePath, `${workDir}/`, copyOpts);
-    core.info(`[INFO] copy ${file}`);
+  } else {
+    await exec.exec('rsync', [
+      '-rptgoDv',
+      '--copy-links',
+      '--copy-dirlinks',
```

Comment:
From `rsync --help`:

```
 -l, --links                 copy symlinks as symlinks
 -L, --copy-links            transform symlink into referent file/dir
     --copy-unsafe-links     only "unsafe" symlinks are transformed
     --safe-links            ignore symlinks that point outside the source tree
     --munge-links           munge symlinks to make them safer (but unusable)
 -k, --copy-dirlinks         transform symlink to a dir into referent dir
 -K, --keep-dirlinks         treat symlinked dir on receiver as dir
 -H, --hard-links            preserve hard links
```

Extended documentation at https://superuser.com/a/1388910

My understanding is that `--copy-links` includes `--copy-dirlinks`, such that you should only need to specify the former.

Related to my comment at https://github.com/peaceiris/actions-gh-pages/issues/103#issuecomment-587646291 and from looking at the rsync docs, I think there are two options for this action:

1. `--copy-links`: dereference all symlinks
2. `--copy-unsafe-links`: only dereference "unsafe" symlinks (symbolic links that point outside the copied tree)

I like option 2 the most because GitHub Pages can handle symlinks that point within the file tree. However, I can imagine users would like access to both modes. Some applications might want to dereference all symlinks while others might want to only dereference unsafe symlinks.

Reply:
Thank you for investigating.

> I like option 2 the most because GitHub Pages can handle symlinks that point within the file tree. However, I can imagine users would like access to both modes. Some applications might want to dereference all symlinks while others might want to only dereference unsafe symlinks.

I have understood the two cases clearly.